src/CallbackController.php

```diff
@@ -31,15 +36,31 @@ public function callback(Request $request)
         if (!$request->has('code')) {
             throw new AuthenticationException('No authorization code received');
         }
-        $token = $this->provider->getAccessToken('authorization_code', [
+        $token = $this->provider->getAccessToken(
+            'authorization_code',
+            [
             'code' => $request->get('code'),
-        ]);
+            ]
+        );
 
         $required = config('oidc-auth.required_claims');
+
+        // attempt to get serialized closure first
+        try {
+            SerializableClosure::setSecretKey(config('app.key'));
+            $unserialized = unserialize($required, ['allowed_classes' => [SerializableClosure::class, Signed::class, Native::class]]);
+
+            if ($unserialized->getClosure() instanceof Closure) {
```

Comment:
doesn't SerializableClosure::getClosure() always returns a Closure? why it this check needed?

Reply:
The idea here is user may pass in data that is malformed or serialized from classes other than `SerializableClosure`.

how about
```suggestion
            if ($unserialized instanceof SerializableClosure) {
```

Reply:
sounds better

Reply:
if not a SerializableClosure, it is a misconfiguration, and we should log it and fail this request, instead of going on in slient